plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveTableHandle.java

```diff
@@ -140,11 +144,13 @@ public HiveTableHandle(
             boolean recordScannedFiles,
             Optional<Long> maxSplitFileSize)
     {
+        checkState(!(partitionNames.isPresent() && partitions.isPresent()), "PartitionInformation cannot be raw and loaded at same time");
```

Comment:
`PartitionInformation` is not "a thing", don't use CamelCase.

- "partition names and partitions list cannot be both present"
- use `isEmpty() || ...isEmpty()` to avoid negation